<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1284</width>
    <height>725</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MLRSv2</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>assets/roboszpon-black.png</normaloff>assets/roboszpon-black.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="6" column="0">
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>5</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="1" column="1">
     <widget class="QLabel" name="logoLabel">
      <property name="text">
       <string>Logo</string>
      </property>
     </widget>
    </item>
    <item row="10" column="1">
     <widget class="QPushButton" name="armButton">
      <property name="text">
       <string>Arm</string>
      </property>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="QComboBox" name="comboBoxDevices"/>
    </item>
    <item row="0" column="1">
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeType">
       <enum>QSizePolicy::Minimum</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>2</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="11" column="1">
     <widget class="QPushButton" name="stopAllButton">
      <property name="text">
       <string>Stop all</string>
      </property>
     </widget>
    </item>
    <item row="6" column="2" rowspan="6">
     <widget class="QTabWidget" name="setpointGroup">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>295</width>
        <height>0</height>
       </size>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="tab_4">
       <attribute name="title">
        <string>Duty</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBox_2">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Setpoint</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_4">
           <item row="0" column="0">
            <widget class="QDoubleSpinBox" name="dutySpinBox">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="minimum">
              <double>-1.000000000000000</double>
             </property>
             <property name="maximum">
              <double>1.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QPushButton" name="dutyButton">
             <property name="text">
              <string>Send</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Current</string>
       </attribute>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Velocity</string>
       </attribute>
       <widget class="QGroupBox" name="groupBox_4">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>10</y>
          <width>179</width>
          <height>70</height>
         </rect>
        </property>
        <property name="title">
         <string>Setpoint</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_6">
         <item row="0" column="1">
          <widget class="QPushButton" name="velocityButton">
           <property name="text">
            <string>Send</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QDoubleSpinBox" name="velocitySpinBox">
           <property name="minimum">
            <double>-100.000000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.500000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Position</string>
       </attribute>
       <widget class="QGroupBox" name="groupBox_5">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>179</width>
          <height>70</height>
         </rect>
        </property>
        <property name="title">
         <string>Setpoint</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_7">
         <item row="0" column="1">
          <widget class="QPushButton" name="positionButton">
           <property name="text">
            <string>Send</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QDoubleSpinBox" name="positionSpinBox">
           <property name="minimum">
            <double>-100.000000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </item>
    <item row="12" column="1">
     <spacer name="verticalSpacer_3">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>1</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="2" column="1">
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>Devices</string>
      </property>
     </widget>
    </item>
    <item row="6" column="1" rowspan="4">
     <widget class="QGroupBox" name="groupBox_3">
      <property name="minimumSize">
       <size>
        <width>180</width>
        <height>0</height>
       </size>
      </property>
      <property name="title">
       <string>Status</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="connectionLabel">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>No device selected.</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="stateLabel">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="flagsLabel">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item row="4" column="1" colspan="2">
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Parameters</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="1" column="1">
        <widget class="QTabWidget" name="tabWidgetParameters"/>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="3" rowspan="13">
     <widget class="QWidget" name="plotView" native="true">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1284</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionCommit_configuratio"/>
    <addaction name="actionRestore_configuratio"/>
    <addaction name="actionFactory_setting"/>
   </widget>
   <widget class="QMenu" name="menuConfiguration">
    <property name="title">
     <string>Flash</string>
    </property>
    <addaction name="actionCommit_configuration"/>
    <addaction name="actionRestore_configuration"/>
    <addaction name="separator"/>
    <addaction name="actionFactory_settings"/>
   </widget>
   <widget class="QMenu" name="menuDevice">
    <property name="title">
     <string>Device</string>
    </property>
    <addaction name="actionSoftware_reset"/>
   </widget>
   <widget class="QMenu" name="menuParameters">
    <property name="title">
     <string>Parameters</string>
    </property>
    <widget class="QMenu" name="menuTemperature">
     <property name="title">
      <string>Temperature</string>
     </property>
     <addaction name="actionOVERHEAT_TEMPERATURE"/>
     <addaction name="actionNO_OVERHEAT_TEMPERATURE"/>
    </widget>
    <widget class="QMenu" name="menuPPID">
     <property name="title">
      <string>PPID</string>
     </property>
     <addaction name="actionPPID_Kp"/>
     <addaction name="actionPPID_Ki"/>
     <addaction name="actionPPID_KD"/>
     <addaction name="actionPPID_deadzone"/>
    </widget>
    <widget class="QMenu" name="menuVPID">
     <property name="title">
      <string>VPID</string>
     </property>
     <addaction name="actionVPID_Kp"/>
     <addaction name="actionVPID_Ki"/>
     <addaction name="actionVPID_Kd"/>
     <addaction name="actionVPID_deadzone"/>
     <addaction name="actionVPID_dUmax"/>
    </widget>
    <widget class="QMenu" name="menuCPID">
     <property name="title">
      <string>CPID</string>
     </property>
     <addaction name="actionENCODER_ZERO"/>
     <addaction name="actionENCODER_FILTER_WINDOW"/>
     <addaction name="actionINVERT_ENCODER"/>
     <addaction name="actionDISABLE_ENCODER_ERRORS"/>
    </widget>
    <widget class="QMenu" name="menuEncoder">
     <property name="title">
      <string>Encoder</string>
     </property>
     <addaction name="actionENCODER_ZERO_2"/>
     <addaction name="actionENCODER_FILTER_WINDOW_2"/>
     <addaction name="actionINVERT_ENCODER_2"/>
     <addaction name="actionDISABLE_ENCODER_ERRORS_2"/>
    </widget>
    <widget class="QMenu" name="menuAXIS">
     <property name="title">
      <string>AXIS</string>
     </property>
     <addaction name="actionAXIS_OFFSET"/>
     <addaction name="actionINVERT_AXIS"/>
    </widget>
    <widget class="QMenu" name="menuCurrent">
     <property name="title">
      <string>Current</string>
     </property>
     <addaction name="actionCURRENT_FeedForward"/>
     <addaction name="actionMIN_CURRENT"/>
     <addaction name="actionMAX_CURRENT"/>
    </widget>
    <widget class="QMenu" name="menuIIR">
     <property name="title">
      <string>IIR</string>
     </property>
     <addaction name="actionIIR_VALUE_CURMEAS"/>
     <addaction name="actionIIR_VALUE_VELMAS"/>
     <addaction name="actionIIR_VALUE_PPIDU"/>
     <addaction name="actionIIR_VALUE_VIPDU"/>
     <addaction name="actionIIR_VALUE_CPIDU"/>
    </widget>
    <widget class="QMenu" name="menuPosition">
     <property name="title">
      <string>Position</string>
     </property>
     <addaction name="actionMIN_POSITION"/>
     <addaction name="actionMAX_POSITION"/>
    </widget>
    <widget class="QMenu" name="menuDuty">
     <property name="title">
      <string>Duty</string>
     </property>
     <addaction name="actionDUTY_DEADZONE"/>
     <addaction name="actionMIN_DUTY"/>
     <addaction name="actionMAX_DUTY"/>
    </widget>
    <widget class="QMenu" name="menuReport">
     <property name="title">
      <string>Report</string>
     </property>
     <addaction name="actionCOMMAND_TIMEOUT"/>
     <addaction name="actionREPORT_RATE"/>
    </widget>
    <widget class="QMenu" name="menuVelocity">
     <property name="title">
      <string>Velocity</string>
     </property>
     <addaction name="actionMIN_VELOCITY"/>
     <addaction name="actionMAX_VELOCITY"/>
    </widget>
    <addaction name="menuTemperature"/>
    <addaction name="menuPPID"/>
    <addaction name="menuVPID"/>
    <addaction name="menuCPID"/>
    <addaction name="menuEncoder"/>
    <addaction name="menuAXIS"/>
    <addaction name="menuCurrent"/>
    <addaction name="menuIIR"/>
    <addaction name="menuDuty"/>
    <addaction name="menuPosition"/>
    <addaction name="menuVelocity"/>
    <addaction name="menuReport"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuConfiguration"/>
   <addaction name="menuDevice"/>
   <addaction name="menuParameters"/>
  </widget>
  <action name="actionConnect_roboszpon">
   <property name="text">
    <string>Connect roboszpon</string>
   </property>
  </action>
  <action name="actionCommit_configuratio">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionRestore_configuratio">
   <property name="text">
    <string>Save as...</string>
   </property>
  </action>
  <action name="actionFactory_setting">
   <property name="text">
    <string>Open...</string>
   </property>
  </action>
  <action name="actionCommit_configuration">
   <property name="text">
    <string>Commit configuration</string>
   </property>
  </action>
  <action name="actionRestore_configuration">
   <property name="text">
    <string>Restore configuration</string>
   </property>
  </action>
  <action name="actionFactory_settings">
   <property name="text">
    <string>Restore factory values</string>
   </property>
  </action>
  <action name="actionSoftware_reset">
   <property name="text">
    <string>Software reset</string>
   </property>
  </action>
  <action name="actionOVERHEAT_TEMPERATURE">
   <property name="text">
    <string>OVERHEAT_TEMPERATURE</string>
   </property>
  </action>
  <action name="actionNO_OVERHEAT_TEMPERATURE">
   <property name="text">
    <string>NO_OVERHEAT_TEMPERATURE</string>
   </property>
  </action>
  <action name="actionPPID_Kp">
   <property name="text">
    <string>PPID_Kp</string>
   </property>
  </action>
  <action name="actionPPID_Ki">
   <property name="text">
    <string>PPID_Ki</string>
   </property>
  </action>
  <action name="actionPPID_KD">
   <property name="text">
    <string>PPID_Kd</string>
   </property>
  </action>
  <action name="actionPPID_deadzone">
   <property name="text">
    <string>PPID_deadzone</string>
   </property>
  </action>
  <action name="actionVPID_Kp">
   <property name="text">
    <string>VPID_Kp</string>
   </property>
  </action>
  <action name="actionVPID_Ki">
   <property name="text">
    <string>VPID_Ki</string>
   </property>
  </action>
  <action name="actionVPID_Kd">
   <property name="text">
    <string>VPID_Kd</string>
   </property>
  </action>
  <action name="actionVPID_deadzone">
   <property name="text">
    <string>VPID_deadzone</string>
   </property>
  </action>
  <action name="actionVPID_dUmax">
   <property name="text">
    <string>VPID_dUmax</string>
   </property>
  </action>
  <action name="actionENCODER_ZERO">
   <property name="text">
    <string>ENCODER_ZERO</string>
   </property>
  </action>
  <action name="actionENCODER_FILTER_WINDOW">
   <property name="text">
    <string>ENCODER_FILTER_WINDOW</string>
   </property>
  </action>
  <action name="actionINVERT_ENCODER">
   <property name="text">
    <string>INVERT_ENCODER</string>
   </property>
  </action>
  <action name="actionDISABLE_ENCODER_ERRORS">
   <property name="text">
    <string>DISABLE_ENCODER_ERRORS</string>
   </property>
  </action>
  <action name="actionENCODER_ZERO_2">
   <property name="text">
    <string>ENCODER_ZERO</string>
   </property>
  </action>
  <action name="actionENCODER_FILTER_WINDOW_2">
   <property name="text">
    <string>ENCODER_FILTER_WINDOW</string>
   </property>
  </action>
  <action name="actionINVERT_ENCODER_2">
   <property name="text">
    <string>INVERT_ENCODER</string>
   </property>
  </action>
  <action name="actionDISABLE_ENCODER_ERRORS_2">
   <property name="text">
    <string>DISABLE_ENCODER_ERRORS</string>
   </property>
  </action>
  <action name="actionAXIS_OFFSET">
   <property name="text">
    <string>AXIS_OFFSET</string>
   </property>
  </action>
  <action name="actionINVERT_AXIS">
   <property name="text">
    <string>INVERT_AXIS</string>
   </property>
  </action>
  <action name="actionCURRENT_FeedForward">
   <property name="text">
    <string>CURRENT_FeedForward</string>
   </property>
  </action>
  <action name="actionMIN_CURRENT">
   <property name="text">
    <string>MIN_CURRENT</string>
   </property>
  </action>
  <action name="actionMAX_CURRENT">
   <property name="text">
    <string>MAX_CURRENT</string>
   </property>
  </action>
  <action name="actionIIR_VALUE_CURMEAS">
   <property name="text">
    <string>IIR_VALUE_CURMEAS</string>
   </property>
  </action>
  <action name="actionIIR_VALUE_VELMAS">
   <property name="text">
    <string>IIR_VALUE_VELMAS</string>
   </property>
  </action>
  <action name="actionIIR_VALUE_PPIDU">
   <property name="text">
    <string>IIR_VALUE_PPIDU</string>
   </property>
  </action>
  <action name="actionIIR_VALUE_VIPDU">
   <property name="text">
    <string>IIR_VALUE_VIPDU</string>
   </property>
  </action>
  <action name="actionIIR_VALUE_CPIDU">
   <property name="text">
    <string>IIR_VALUE_CPIDU</string>
   </property>
  </action>
  <action name="actionDUTY_DEADZONE">
   <property name="text">
    <string>DUTY_DEADZONE</string>
   </property>
  </action>
  <action name="actionMAX_DUTY">
   <property name="text">
    <string>MAX_DUTY</string>
   </property>
  </action>
  <action name="actionMIN_DUTY">
   <property name="text">
    <string>MIN_DUTY</string>
   </property>
  </action>
  <action name="actionMIN_POSITION">
   <property name="text">
    <string>MIN_POSITION</string>
   </property>
  </action>
  <action name="actionMAX_POSITION">
   <property name="text">
    <string>MAX_POSITION</string>
   </property>
  </action>
  <action name="actionMIN_VELOCITY">
   <property name="text">
    <string>MIN_VELOCITY</string>
   </property>
  </action>
  <action name="actionMAX_VELOCITY">
   <property name="text">
    <string>MAX_VELOCITY</string>
   </property>
  </action>
  <action name="actionCOMMAND_TIMEOUT">
   <property name="text">
    <string>COMMAND_TIMEOUT</string>
   </property>
  </action>
  <action name="actionREPORT_RATE">
   <property name="text">
    <string>REPORT_RATE</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
